app/src/main/java/com/naman14/timberx/sdl/SdlService.kt

```diff
@@ -66,6 +69,14 @@ class SdlService : Service(), KoinComponent {
     private val genreRepository by inject<GenreRepository>()
     private val playlistRepository by inject<PlaylistRepository>()
 
+    val broadCastReceiver = object : BroadcastReceiver() {
+        override fun onReceive(contxt: Context?, intent: Intent?) {
+            if (intent != null && Constants.ACTION_NOW_PLAYING.equals(intent.action)){
+                onNowPlaying(intent.extras)
+            }
+        }
+    }
```

Comment:
```suggestion
    val broadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            if (intent == null || sdlManager == null || !ready) {
                return
            }
            when (intent.action) {
                ACTION_NOW_PLAYING -> {
                    intent.extras?.let { onNowPlaying(it) }
                }
            }
        }
    }
```
Some style change suggestions based on my own implementation